pkg/frontend/querymiddleware/astmapper/sharding_test.go

```diff
@@ -503,6 +503,16 @@ func TestShardSummer(t *testing.T) {
 			out:                    `sum(` + concatShards(3, `sum(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `) > (sum(` + concatShards(3, `sum(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `) / sum(` + concatShards(3, `count(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `))`,
 			expectedShardedQueries: 9,
 		},
+		{
+			in:                     `group by (a, b) (metric)`,
+			out:                    `group by (a, b) (` + concatShards(3, `group by (a, b) (metric{__query_shard__="x_of_y"})`) + `)`,
+			expectedShardedQueries: 3,
+		},
+		{
+			in:                     `count by (a) (group by (a, b) (metric))`,
+			out:                    `count by (a) (group by (a, b) (` + concatShards(3, `group by (a, b) (metric{__query_shard__="x_of_y"})`) + `))`,
+			expectedShardedQueries: 3,
+		},
```

Comment:
[nit] Could you also add this test case, please?

```
in:                     `count(group without () ({namespace="foo"}))`,
			out:                    `count(group without() (` + concatShards(3, `group without() ({namespace="foo",__query_shard__="x_of_y"})`) + `))`,
			expectedShardedQueries: 3,
```

Reply:
Added in https://github.com/grafana/mimir/pull/6024/commits/b1f888f392f0b2511b2fca9ca70fceb86a10b36a